packages/terra-disclosure-manager/docs/README.md

```diff
@@ -0,0 +1,209 @@
+|`goBack()`|Allows a component to remove itself from the disclosure stack. Functionally similar to `dismiss`, however `onBack` is only provided to components in the stack that have a previous sibling. This is generally integrated into face-up disclosure controls as a Back button or similar.|
+|`maximize()`|Allows a component to maximize its presentation size. This is only provided if the component is not already `maximize`-d|
+|`minimize()`|Allows a component to minimize its presentation size. This is only provided if the component is currently `maximize`-d|
+|`requestFocus()`|Allows a component to request focus from the disclosure in the event that the disclosure mechanism in use utilizes a focus trap. This can be integrated with the Popup and similar focus-stealing controls.|
+|`requestFocus()`|Allows a component to release focus from itself and return it to the disclosure. This can be integrated with the Popup and similar focus-stealing controls.|
+|`registerDismissCheck(func)`|Allows a component to register a function with the DisclosureManager that will be called before the component is dismissed for any reason.|
+
+Each of these functions returns a Promise that can be used for chaining, if necessary.
+
+##### A Note on `registerDismissCheck`
+
+The function given to registerDismissCheck should return a resolved or rejected Promise. If the Promise is resolved, the component is guaranteed to be dismissed. If cleanup logic needs to execute before the component is dismissed, it is a good idea to execute before returning the resolved Promise. If a rejected Promise is returned, the component will not be dismissed. Components can render a prompt or confirmation window to give users control over the dismissal, if desired.
+
+Example:
+```javascript
+// MyDisclosedComponent.jsx
+
+componentDidMount() {
+  const { app } = this.props;
+
+  // A dismiss function can be registered multiple times, but it usually makes sense to
+  // register a bound function a single time on mount.
+  app.registerDismissCheck(this.unsavedChangesCheck);
+}
+
+unsavedChangesCheck() {
+  const { store } = this.props;
+  const { hasUnsavedChanges } = this.state;
+
+  return new Promise((resolve, reject) => {
+    if (hasUnsavedChanges) {
+      // The dismissal could be outright denied.
+      // reject();
+
+      // Or, a confirmation could allow user control.
+      if (!confirm('MyDisclosedComponent has unsaved changes that will be lost. Do you wish to continue?')) {
+        reject();
+        return;
+      }
+
+      // Otherwise, resolving the Promise will cause the component to be dismissed and unmounted. Any cleanup needed
+      // before dismissal should occur before the Promise is resolved.
+      store.dispatch(myCleanupActionOrSomething());
+
+      resolve();
+    }
+  });
+}
+```
+
+### Example
+
+```jsx
+// MyDisclosureManagerImpl.jsx
```

Comment:
If we want consumers to easily use this component, it is good to have a full working example which can be copy / pasted by consumers here. The modal disclosure example after the disclosure Object API docs is a good start to a working example.

Reply:
I think the best example of the disclosure manager working would be the modal manager. We should point them there to look at that code. I'm not sure how many consumers will directly create another 'x type disclosure manager' outside of terra itself.

Reply:
I agree that an fully working example included here could hurt more than help. Most consumers will not implement a DisclosureManager instance directly, and a working (yet un-opinionated) example included here would still be substandard in terms of actual usability. 

I can link out to the Modal-/SlidePanelManager for more reference, which will be more useful to consumers. If I do that, should this simple example be removed altogether? Or does seeing the props spelled out in their natural environment like this add just enough value to stick around?

Reply:
I do think referencing to Modal-/SlidePanelManager would be good, however, I think providing a simpler example may be beneficial as well. The examples on site use every possible prop options, with little distinguishing difference between the instances, which can be overwhelming.

Reply:
> Most consumers will not implement a DisclosureManager instance directly

Are you saying this module should never be used by a module outside of terra-framework?

Reply:
Most likely yes, We'd expect consumers to use the modal manager or the slide panel manager. This is an abstraction that allows the guts of those components to be common. That said, consumers could conceivably use this to create their own special disclosure manager and we will still release this as it's own consumable package, as the slide panel and modal managers are.